Potential remove fix and comprehensive CutNode tests!

RyanR712 opened the pull request:
Looking for review ASAP

Commit:
Potential remove fix. Looking for review
CutNode.ts - replaced existing remove method with new attempt and updated documentation a bit

Commit:
Remove method changes and test bundle!
CutNode.test.ts - added from branch 127-cutnode-tests, edited some Points in AtomNode constructions with the understanding that they now use the bottom left vertex
CutNode.ts - remove now functions closer to what is expected. Passes 6 of the existing 7 remove tests, skips the 5th in the remove section
Ellipse.test.ts - brought changes over from 127-cutnode-tests

Commit:
Removed an unnecessary comment

Commit:
Merge branch 'master' into 157-remove-method-fix

DawnTheWitch reviewed (approved):
I did my best to read through it as best as I could. To me this makes sense and should work. I spent way too long rereading it to make sure I understood it.

AnushaTiwari5 commented on src/AEG/CutNode.ts:
```diff
@@ -123,36 +123,48 @@ export class CutNode {
-            if (isSmallest) {
-                //What to do to remove this node?
-                return true;
-            }
         }
-
         return false;
     }
 
```
What do you do if the point is contained in the cut, but not in any children of the cut/the cut doesn't have any children?

DawnTheWitch replied:
I might be misunderstanding as well but wouldn't this solve that? And for when the cut doesn't have any children lines 137-143 handle that
![image](https://github.com/RAIRLab/Peirce-My-Heart/assets/93628226/6066a6c7-ca85-4917-adf7-530e629bb5ac)

AnushaTiwari5 replied:
Aren't lines 136 - 166 the for loop with stuff happening if any child contain the point? I was wondering what happen if we don't enter that loop/exit that loop without any children containing the point. 
For example, in a situation like in the ss below, the point is within the cut but not within any of the children. So if remove is called on the cut, it would pass through the for loop without anything happening because the if on line 137 would return false
![image](https://github.com/RAIRLab/Peirce-My-Heart/assets/89546165/22fa8953-5b02-4176-94ec-f389b9026502)

DawnTheWitch replied:
I'm pretty sure with what we talked about this function was meant to be used on the parent of the item you want to delete, that way you can access the to be deleted child easier. But that does also make me realize that 131-133 shouldn't include logic to remove the sheet since there's really no way to remove the item itself here

AnushaTiwari5 replied:
Ohh that does make more sense then, thank you. I think we do need the ellipse === null check to avoid instance of TS screaming at us that ellipse might be null. That happens if you sometimes if you don't explicitly check. I'm not sure about the containsPoint check though, because ideally wouldn't we always call this assuming it does contain the point?

Commit:
Added new test and removed condition
CutNode.ts - cleared up documentation and removed a condition that would, in practice, prevent the sheet from removing any of its children
CutNode.test.ts - Added a test for this above case that passes. Decently comprehensive, if I say so myself

Commit:
Merge branch '157-remove-method-fix' of https://github.com/RAIRLab/Peirce-My-Heart into 157-remove-method-fix
